### PR TITLE
Week 8

### PR DESCRIPTION
**Pull request by jepetko**
- 1 Assignment: Create subscriptions
- 2 Assignment: Admin views payments
- 3 Assignment: Handle payment failures
- 10    Assignment: Search Video by Title
- 11    Assignment: Search Video by Title and Description
- 12    Assignment: Build Out the Entire Workflow for Advanced Search
- 13    Assignment: Expand Search into Reviews
- 14    Assignment: Filter with Average Rating
- 15    Assignment: Build Out Advanced Search Options

### CODE REVIEW

File: app/decorators/payment_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class PaymentDecorator < Draper::Decorator
+  delegate_all
+
+  def amount
+    amount = read_attribute :amount
+    formatted_amount = "%.2f" % (amount.to_f/100)
+    "#{formatted_amount} €"
```
Comment (tomtomecek) [inline app/decorators/payment_decorator.rb]: I recommend you to check - [number_to_currency](http://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_currency) helper.

File: app/models/stripe_wrapper.rb [hunk 2]
```diff
@@ -12,15 +18,54 @@ def successful?
...
       begin
         Stripe::Charge.create(card: options[:card],
                               amount: 999,
                               description: options[:description],
-                              currency: 'usd')
+                              currency: 'eur')
         new :success
       rescue Stripe::CardError => err
-        new :card_error, error_type: err.class.name, error_message: err.message
+        handle_exception err
```
Comment (tomtomecek) [inline app/models/stripe_wrapper.rb:44]: I like this extraction! :) :+1:

File: spec/fabricators/review_fabricator.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Fabricator(:review) do
+  video nil
```
Comment (tomtomecek) [inline spec/fabricators/review_fabricator.rb]: I think you can take out `video` if you don't want it to be assigned here.

File: spec/support/capybara_helper.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 module CapybaraHelper
-  def sign_in(user)
+  def sign_in(user = nil)
+    user = Fabricate(:user) if !user
```
Comment (tomtomecek) [inline spec/support/capybara_helper.rb]: `user ||= Fabricate(:user)` is more succinct.

File: app/decorators/payment_decorator.rb (hunk 1 above)
Comment (jepetko) [inline app/decorators/payment_decorator.rb]: fixed. I had to include  ActionView::Helpers::NumberHelper in the decorator to do this. Furthermore, it had to create a locale file.

File: spec/support/capybara_helper.rb (hunk 4 above)
Comment (jepetko) [inline spec/support/capybara_helper.rb]: fixed

File: spec/fabricators/review_fabricator.rb (hunk 3 above)
Comment (jepetko) [inline spec/fabricators/review_fabricator.rb]: fixed.

**Pull request merged**